### PR TITLE
fix(kubernetes/serverGroup): use the actual server group manager's kind in name

### PR DESCRIPTION
**Pull request by ncknt**
Prior to this change, the Kubernetes server group manager's name was hardcoded with prefix `deployment`. We're now looking using the kind as it comes from the server (`$kind $name`).

### CODE REVIEW

File: app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      } as any;
+    };
+
+    const sg = await transformer.normalizeServerGroup(sgAccount, app);
+    expect(sg).not.toBe(null);
+    expect(sg.serverGroupManagers.length).toBe(1);
+    expect(sgMgrRef.name).toBe('daemonSet sgmanager');
+  });
+
+  it('normalized server group different account', async () => {
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts:38]: it's semi-conventional to try to write these such that the test description makes a sentence with `it` e.g. `it('normalizes the server group name')`

File: app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+describe('KubernetesV2ServerGroupTransformer', function() {
+  it('normalized server group name', async () => {
+    const transformer = new KubernetesV2ServerGroupTransformer();
+    const ACCOUNT = 'myaccount';
+    const LOCATION = 'myns';
+    const SG_MANAGER = 'sgmanager';
+    const KIND = 'daemonSet';
+
+    const sgMgrRef = { account: ACCOUNT, location: LOCATION, name: SG_MANAGER };
+    const sgAccount: IKubernetesServerGroup = ({
```
Comment (danielpeach) [inline app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts:16]: Can drop the left-hand side `IKubernetesServerGroup` typing unless you think it's more readable that way.

**Review by danielpeach — APPROVED** [review]
lgtm

File: app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts (hunk 1 above)
Comment (ncknt) [reply app/scripts/modules/kubernetes/src/serverGroup/serverGroupTransformer.service.spec.ts]: The `d` is too close to the `s` :) Thanks!

**Pull request merged**